plugins/destination/duckdb/client/write.go

```diff
@@ -27,30 +31,29 @@ func nonPkIndices(sc *arrow.Schema) []int {
 // As a workaround, we delete the row and insert it again. This makes it non-atomic, unfortunately,
 // but this is unavoidable until support is added to duckdb itself.
 // See https://github.com/duckdb/duckdb/blob/c5d9afb97bbf0be12216f3b89ae3131afbbc3156/src/storage/table/list_column_data.cpp#L243-L251
-func containsList(sc *arrow.Schema) bool {
-	for _, f := range sc.Fields() {
+func containsList(sc *schema.Table) bool {
```

Comment:
we need to check for other list types as well (large list, fixed size list).
`arrow.IsListLike` (+ possibly check that the type ID isn't `arrow.MAP`, if we have special handling for those)